seismicpro/src/seismic_metrics.py

```diff
@@ -0,0 +1,135 @@
+            metrics = metrics_storage.copy()
+        metrics = np.array(metrics, dtype=np.float32)
+
+        if isinstance(agg_bins_fn, str):
+            call_agg_bins = getattr(NumbaNumpy, agg_bins_fn)
+            call_agg_bins = call_agg_bins(None, **agg_bins_kwargs) if agg_bins_kwargs is not None else call_agg_bins
+        elif callable(agg_bins_fn):
+            call_agg_bins = agg_bins_fn
+        else:
+            raise ValueError('agg_bins_fn should be whether str or callable, not {}'.format(type(agg_bins_fn)))
+
+        metric_map = self.construct_metrics_map(coords_x=coords_x, coords_y=coords_y,
+                                                metrics=metrics, bin_size=bin_size,
+                                                agg_bins_fn=call_agg_bins)
+
+        if plot:
+            extent = [coords_x.min(), coords_x.max(), coords_y.min(), coords_y.max()]
+            plot_metrics_map(metrics_map=metric_map, extent=extent, cm=cm, title=title,
+                             figsize=figsize, save_to=save_to, dpi=dpi, pad=pad,
+                             **plot_kwargs)
+        return metric_map
+
+    @staticmethod
+    @njit(parallel=True)
+    def construct_metrics_map(coords_x, coords_y, metrics, bin_size, agg_bins_fn):
+        """njit map"""
+        bin_size_x, bin_size_y = bin_size
+        range_x = np.arange(coords_x.min(), coords_x.max() + 1, bin_size_x)
+        range_y = np.arange(coords_y.min(), coords_y.max() + 1, bin_size_y)
+        metrics_map = np.full((len(range_y), len(range_x)), np.nan)
+        for i in prange(len(range_x)): #pylint: disable=not-an-iterable
+            for j in prange(len(range_y)): #pylint: disable=not-an-iterable
+                mask = ((coords_x - range_x[i] >= 0) & (coords_x - range_x[i] < bin_size_x) &
+                        (coords_y - range_y[j] >= 0) & (coords_y - range_y[j] < bin_size_y))
+                if mask.sum() > 0:
+                    metrics_map[j, i] = agg_bins_fn(np.ravel(metrics[mask]))
+        return metrics_map
+
+
+def quantile(array, q):
+    """ numba quantile. """
+    _ = array
+    return  njit(lambda array: np.quantile(array, q=q))
+
+def absquantile(array, q):
+    """ numba absquantile. """
+    _ = array
+    return njit(lambda array: _absquantile(array, q=q))
+
+@njit
+def _absquantile(array, q):
+    shifted_array = array - np.mean(array)
+    val = np.quantile(np.abs(shifted_array), q)
+    ind = np.argmin(np.abs(np.abs(shifted_array) - val))
+    return array[ind]
+
+class NumbaNumpy:
+    """ Holder for jit-accelerated functions. """
+    #pylint: disable = unnecessary-lambda, undefined-variable
+    nanmin = njit(lambda array: np.nanmin(array))
+    nanmax = njit(lambda array: np.nanmax(array))
+    nanmean = njit(lambda array: np.nanmean(array))
+    nanstd = njit(lambda array: np.nanstd(array))
+
+    min = nanmin
+    max = nanmax
+    mean = nanmean
+    std = nanstd
+    quantile = quantile
```

Comment:
```suggestion
    def quantile(q): return  njit(lambda array: np.quantile(array, q=q))
```

And don't pass `None` to `call_agg_bins` when creating it with kwargs

Reply:
thank for the tip